cpu/msp430_common/startup.c

```diff
@@ -26,6 +26,8 @@
 
 extern void board_init(void);
 
+#define STACK_EXTRA 32
```

Comment:
`STACK_EXTRA` is simply a margin to leave some space on the stack for the calls of `kernel_init` and further `cpu_switch_context_exit`. It  ensures that the stack pointer doesn't decrease to `__heap_end` before it is set to any thread stack in `kernel_init`.

Reply:
```suggestion
/* Leave some extra space in the stack to allows us to finish the kernel initialization procedure.
 * __heap_end is set the current stack, minus STACK_EXTRA since there is still code to execute.
 */
#define STACK_EXTRA 32
```

Reply:
Changed.